src/app/cooling_device.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 Turing Machines
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::path::Path;
+
+use serde::Serialize;
+
+#[derive(Debug, Serialize)]
+pub struct CoolingDevice {
+    pub device: String,
+    pub speed: u8,
+    pub max_speed: u8,
+}
+
+pub async fn get_cooling_state() -> Vec<CoolingDevice> {
+    let mut result = Vec::new();
+
+    if let Ok(mut dir) = tokio::fs::read_dir("/sys/class/thermal").await {
```

Comment:
I must filter here for `/sys/class/thermal/cooling_device[0-*]`.

In our v2.4 boards, we currently only have `cooling_device0` inside `thermal`, but in theory and according to https://www.kernel.org/doc/Documentation/thermal/sysfs-api.txt, there could be `thermal_zone[0-*]`, which I don't want to iterate over here.

Reply:
Done @ [12a0cbf](https://github.com/turing-machines/bmcd/pull/84/commits/12a0cbf01bba7953a7f20025580ba283a71050f6)